lib/libzpool/uzfs_rebuilding.c

```diff
@@ -219,8 +230,7 @@ uzfs_get_io_diff(zvol_state_t *zv, blk_metadata_t *low,
 
 			lun_offset += zv->zv_metavolblocksize;
 		}
-
-		if (diff_count) {
+		if (!ret && diff_count) {
```

Comment:
Not related to this PR but I think there is memory leak i.e. memory associated with snap_zv might be leaked. Given below is the function call sequence:-
+uzfs_get_io_diff
    +get_snapshot_zv
          +uzfs_hold_dataset (line 94) can error out, in that case we might leak memory associated with snap_zv.

Reply:
done uzfs_rele_dataset wherever open_dataset is success, and, hold_dataset is failure

Reply:
done uzfs_close_dataset wherever open_dataset is success, and, hold_dataset is failure